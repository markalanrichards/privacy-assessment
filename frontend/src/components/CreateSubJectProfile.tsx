import * as React from "react";

import {
  SubjectProfileAvro,
  SubjectProfileCreateAvro,
  SubjectProfileUpdateAvro
} from "../interfaces/SubjectProfile";

interface PropsInterface {
  mutate(...SubjectProfileCreateAvro): void;
}

export class CreateSubJectProfile extends React.Component<
  PropsInterface,
  SubjectProfileCreateAvro
> {
  public state = {
    customerProfileId: "",
    externalSubjectName: "",
    externalSubjectReference: ""
  };

  public reset = () =>
    this.setState({
      customerProfileId: "",
      externalSubjectName: "",
      externalSubjectReference: ""
    });

  public updateField = stateKey => e => {
    this.setState({ [stateKey]: e.target.value });
  };

  public handelSubmit = e => {
    e.preventDefault();
    return (
      this.props.mutate({
        variables: {
          customerProfileId: this.state.customerProfileId,
          externalSubjectName: this.state.externalSubjectName,
          externalSubjectReference: this.state.externalSubjectReference
        }
      }),
      this.reset()
    );
  };

  public render() {
    return (
      <div>
        <form onSubmit={this.handelSubmit}>
          <div className="group">
            <label>customerProfileId</label>
            <input
              type="text"
              id="customerProfileId"
              value={this.state.customerProfileId}
              onChange={this.updateField("customerProfileId")}
            />
          </div>
          <div className="group">
            <label>externalSubjectName</label>
            <input
              type="text"
              id="externalSubjectName"
              value={this.state.externalSubjectName}
              onChange={this.updateField("externalSubjectName")}
            />
          </div>
          <div className="group">
            <label>externalSubjectReference</label>
            <input
              type="text"
              id="externalSubjectReference"
              value={this.state.externalSubjectReference}
              onChange={this.updateField("externalSubjectReference")}
            />
          </div>
          <button data-testid="submit-button" type="submit">
            Submit
          </button>
          <button data-testid="reset-button" type="button" onClick={this.reset}>
            Reset
          </button>
        </form>
      </div>
    );
  }
}
