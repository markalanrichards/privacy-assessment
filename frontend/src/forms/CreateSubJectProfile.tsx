import { useState } from 'react';

export function CreateSubJectProfile({ mutation }) {
  const [customerProfileId, setCustomerProfileId] = useState('');
  const [externalSubjectName, setExternalSubjectName] = useState('');
  const [externalSubjectReference, setExternalSubjectReference] = useState('');

  return (
    <div>
      <form
        onSubmit={(e) => {
          e.preventDefault();
          mutation.mutate({
            customerProfileId,
            externalSubjectName,
            externalSubjectReference,
          });
        }}
      >
        <div className="group">
          <label htmlFor="customerProfileId">customerProfileId</label>
          <input
            type="text"
            id="customerProfileId"
            value={customerProfileId}
            onChange={(e) => setCustomerProfileId(e.target.value)}
          />
        </div>
        <div className="group">
          <label htmlFor="externalSubjectName">externalSubjectName</label>
          <input
            type="text"
            id="externalSubjectName"
            value={externalSubjectName}
            onChange={(e) => setExternalSubjectName(e.target.value)}
          />
        </div>
        <div className="group">
          <label htmlFor="externalSubjectReference">
            externalSubjectReference
          </label>
          <input
            type="text"
            id="externalSubjectReference"
            value={externalSubjectReference}
            onChange={(e) => setExternalSubjectReference(e.target.value)}
          />
        </div>
        <button
          data-testid="submit-button"
          type="submit"
          disabled={
            !Boolean(
              customerProfileId.length &&
                externalSubjectName.length &&
                externalSubjectReference.length
            )
          }
        >
          Submit
        </button>
        <button
          data-testid="reset-button"
          type="button"
          onClick={() => {
            setCustomerProfileId('');
            setExternalSubjectName('');
            setExternalSubjectReference('');
          }}
        >
          Reset
        </button>
      </form>
    </div>
  );
}
