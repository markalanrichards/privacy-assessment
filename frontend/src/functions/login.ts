export interface LocalStorageFN<T> {
  setItem(index: string, item: T): void;
  getItem(index: string): T;
  clear(): void;
}
interface LoginInterface {
  username: string;
  status: string;
  login: boolean;
  error: null;
}
export const login = (localStorageFn: LocalStorageFN<string>) => (
  username: string
): LoginInterface => {
  localStorageFn.setItem("username", username);
  return {
    username: localStorageFn.getItem("username"),
    status: "success",
    login: true,
    error: null
  };
};

interface LogoutInterface {
  status: string;
  login: boolean;
  error: null;
}

export const logout = (
  localStorageFn: LocalStorageFN<string>
): LogoutInterface => {
  localStorageFn.clear();
  return {
    status: "logged out",
    login: false,
    error: null
  };
};

interface GetUserInterface {
  username: string;
  status: string;
  login: boolean;
  error: null;
}

export const getUser = (
  localStorageFn: LocalStorageFN<string>
): GetUserInterface => {
  const isTheUserLogedIn = localStorageFn.getItem("username");
  return isTheUserLogedIn === null || "undefined"
    ? {
        username: null,
        status: "No user logged in",
        login: false,
        error: null
      }
    : {
        username: isTheUserLogedIn,
        status: "User is logged in",
        login: true,
        error: null
      };
};
