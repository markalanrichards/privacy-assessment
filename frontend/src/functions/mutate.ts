interface MutateInterface<T> {
  externalLegalName: string;
  externalEmail: string;
}

export const mutate = (variables: MutateInterface<string>) => {
  return Promise.resolve({
    variables: {
      externalLegalName: variables.externalLegalName,
      externalEmail: variables.externalEmail
    }
  });
};
