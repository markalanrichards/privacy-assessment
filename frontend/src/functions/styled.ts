import * as styledComponents from "styled-components";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

interface Color {
  primaryColor: string;
}

interface Size {
  large: string;
}

export interface IThemeInterface {
  color: Color;
  size: Size;
}

export const defaultTheme: IThemeInterface = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

export default styled;
export { css, injectGlobal, keyframes, ThemeProvider };
