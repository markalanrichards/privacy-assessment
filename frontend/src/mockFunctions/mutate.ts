import {
  SubjectProfileAvro,
  SubjectProfileCreateAvro,
  SubjectProfileUpdateAvro,
} from '../interfaces/SubjectProfile';

export const mutate = (
  variables:
    | SubjectProfileAvro
    | SubjectProfileCreateAvro
    | SubjectProfileUpdateAvro
) => {
  return Promise.resolve({
    variables,
  });
};
