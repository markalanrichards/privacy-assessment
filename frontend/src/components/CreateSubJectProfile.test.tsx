import * as React from "react";
import { CreateSubJectProfile } from "./CreateSubJectProfile";
import { shallow } from "enzyme";
import * as mockApi from "../functions/mutate";

const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  const input = wrapper.find(inputSelector);
  input.simulate("change", {
    target: { value: newValue }
  });
  return wrapper.find(inputSelector);
};
describe("Form", () => {
  beforeEach(() => jest.resetAllMocks());
  test("lets me fill out the form with values", () => {
    const wrapper = shallow(<CreateSubJectProfile mutate={mockApi.mutate} />);
    expect(wrapper.find("#customerProfileId").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectName").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectReference").props().value).toEqual("");
    const customerProfileIdInput = simulateChangeOnInput(
      wrapper,
      "#customerProfileId",
      "james"
    );
    const externalSubjectNameInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectName",
      "james"
    );

    const externalSubjectReferenceInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectReference",
      "james"
    );

    expect(customerProfileIdInput.props().value).toEqual("james");
    expect(externalSubjectNameInput.props().value).toEqual("james");
    expect(externalSubjectReferenceInput.props().value).toEqual("james");
    expect(wrapper).toMatchSnapshot();
  });

  test("lets me click the reset button to clear the form", () => {
    const wrapper = shallow(<CreateSubJectProfile mutate={mockApi.mutate} />);
    expect(wrapper.find("#customerProfileId").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectName").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectReference").props().value).toEqual("");
    const customerProfileIdInput = simulateChangeOnInput(
      wrapper,
      "#customerProfileId",
      "james"
    );
    const externalSubjectNameInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectName",
      "james"
    );

    const externalSubjectReferenceInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectReference",
      "james"
    );
    const resetBtn = wrapper.find('button[data-testid="reset-button"]');
    resetBtn.simulate("click");
    expect(wrapper.find("#customerProfileId").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectName").props().value).toEqual("");
    expect(wrapper.find("#externalSubjectReference").props().value).toEqual("");
    expect(wrapper).toMatchSnapshot();
  });

  test("data is sumbited when button is pressed", () => {
    jest
      .spyOn(mockApi, "mutate")
      .mockImplementation(() => Promise.resolve({ message: "Saved user!" }));
    const wrapper = shallow(<CreateSubJectProfile mutate={mockApi.mutate} />);

    const customerProfileIdInput = simulateChangeOnInput(
      wrapper,
      "#customerProfileId",
      "james"
    );
    const externalSubjectNameInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectName",
      "james"
    );

    const externalSubjectReferenceInput = simulateChangeOnInput(
      wrapper,
      "#externalSubjectReference",
      "james"
    );

    wrapper.find("form").simulate("submit", {
      preventDefault: () => {}
    });

    expect(mockApi.mutate).toHaveBeenCalledWith({
      variables: {
        customerProfileId: "james",
        externalSubjectName: "james",
        externalSubjectReference: "james"
      }
    });
  });
});
