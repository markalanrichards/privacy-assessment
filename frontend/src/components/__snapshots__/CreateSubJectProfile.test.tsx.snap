// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form lets me click the reset button to clear the form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label>
        customerProfileId
      </label>
      <input
        id="customerProfileId"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="group"
    >
      <label>
        externalSubjectName
      </label>
      <input
        id="externalSubjectName"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="group"
    >
      <label>
        externalSubjectReference
      </label>
      <input
        id="externalSubjectReference"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Submit
    </button>
    <button
      data-testid="reset-button"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Form lets me fill out the form with values 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="group"
    >
      <label>
        customerProfileId
      </label>
      <input
        id="customerProfileId"
        onChange={[Function]}
        type="text"
        value="james"
      />
    </div>
    <div
      className="group"
    >
      <label>
        externalSubjectName
      </label>
      <input
        id="externalSubjectName"
        onChange={[Function]}
        type="text"
        value="james"
      />
    </div>
    <div
      className="group"
    >
      <label>
        externalSubjectReference
      </label>
      <input
        id="externalSubjectReference"
        onChange={[Function]}
        type="text"
        value="james"
      />
    </div>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Submit
    </button>
    <button
      data-testid="reset-button"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;
